src/main/java/org/myrobotlab/service/abstracts/AbstractSpeechSynthesis.java

```diff
@@ -41,6 +41,17 @@
    * synthesis service
    */
   Map<String, String> substitutions = new HashMap<String, String>();
+  
+  /**
+   * generalized list of languages and their codes - if useful
+   */
+  protected Map<String, Locale> locales = new HashMap<>();
+  
+
+  /**
+   * the current locale this service is set to e.g. en-US it-IT fr etc...
+   */
+  protected Locale locale;

```

Comment:
i don't think this should be here.. the subclasses should be implementing the interfaces for LocaleProvider.  the subclass should be able to determine if it needs a locale object locally or not.

Reply:
I think your getting confused - subclasses should simply create their own getLocales - ie give a list of locales they support.  The internals of setLocale and now that I think of it sending an error if setLocale is outside the set of getLocales.  It would be a shame and a mess to duplicate this code in each service.